nautobot_device_lifecycle_mgmt/models.py

```diff
@@ -212,6 +213,46 @@ def clean(self):
     objects = ValidatedSoftwareLCMQuerySet.as_manager()
 
 
+@extras_features(
+    "graphql",
+)
+class DeviceHardwareNoticeResult(PrimaryModel):
+    """Device hardware notice details model."""
+
+    device = models.OneToOneField(
+        to="dcim.Device",
+        on_delete=models.CASCADE,
+        help_text="The device",
+        blank=False,
+        related_name="device_hardware_notice",
+    )
+    hardware_notice = models.ForeignKey(
+        to="nautobot_device_lifecycle_mgmt.HardwareLCM",
+        on_delete=models.CASCADE,
+        help_text="Device hardware notice",
+        null=True,
+        blank=True,
+        related_name="+",
```

Comment:
It might be useful to be able to get relevant Notice Results from HardwareLCM object, what do you think?

Reply:
In general, I prefer relationships to exist unless there is a conflict preventing it, you never know what you'll need in the future! However, the other "Result" models have ` related_name="+"` so I stuck with that for consistency. Do you know if there is a reason for this on the other models?

Reply:
I think the other models lack those for historical reasons mostly. Going forward I think it'd be best to include reverse relationships unless there's a specific reasons not to do it.

Reply:
I agree. The model has been updated and I also updated the migration so we don't end up with 2 (or more) after this PR is merged.